### DISCUSSION TITLE
[Bug]: Responsible Disclosure of Potential Security Vulnerabilities

### DISCUSSION THREAD

**Discussion by DeenoB** [danny-avila/LibreChat#3315] [Troubleshooting] [answered]
### What happened?

The AppSec team at [REA Group](https://www.rea-group.com/about-us/about-rea-group/) have performed a penetration test of LibreChat and have discovered a number of security vulnerabilities. We would like to work with the maintainer of LibreChat to discuss impact, remediation, and whether they should be raised as CVEs. 

We have opted to use communication method Option 2 of your [Security Policy](https://github.com/danny-avila/LibreChat/security) (raising this issue). Trying to create a security issue in the repo only links to the security policy and we can't raise an advisory from there.

Please advise for the best way to engage you; ideally an email address (is contact@librechat.ai still appropriate?). We would prefer not to use Discord for this communication. Thanks team!

### Steps to Reproduce

To be discussed in private channels.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
mailto:contact@librechat.ai Works, thank you

**Comment by R-Prady** [comment]
Could you please highlight the security issues?

**Reply by DeenoB** [reply]
Hi @R-Prady, I emailed Danny in private and it seems like he has merged fix code for the issues.
I'll continue to discuss with him to see if he would like us to disclose them publicly and/or raise CVEs for them. I'd prefer to hold off until a fix version has been released which contains the fixes. Cheers!

**Comment by DeenoB** [comment]
Hi all, two CVEs have been reserved for the vulnerabilities raised:

- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-41703
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-41704

@danny-avila FYI, we will contact you further by email to discuss the fixes you merged recently (#3363). Thanks!

**Comment by DeenoB** [comment, marked as answer]
Hi all, full write-ups of the vulnerabilities have been published:

- [CVE-2024-41703](https://github.com/realestate-com-au/vulnerability-disclosures/blob/main/LibreChat/CVE-2024-41703.md)
- [CVE-2024-41704](https://github.com/realestate-com-au/vulnerability-disclosures/blob/main/LibreChat/CVE-2024-41704.md)

Please update to at least v0.7.4 as both issues have been fixed in that version. Happy to close this discussion.

Thanks again @danny-avila for all your help in getting these merged!
